cmd/fleet/serve.go

```diff
@@ -426,27 +434,48 @@ the way that the Fleet server works.
 					initFatal(err, "validate authentication with Apple APNs certificate")
 				}
 				cancel()
+			}
 
-				if config.MDMApple.Enable {
-					if err := config_apple.VerifyDEP(config.MDMApple); err != nil {
-						initFatal(err, "verify apple mdm DEP config")
-					}
-					scepStorage, err = mds.NewMDMAppleSCEPDepot(appleSCEPCertPEM, appleSCEPKeyPEM)
-					if err != nil {
-						initFatal(err, "initialize mdm apple scep storage")
-					}
-					mdmStorage, err = mds.NewMDMAppleMDMStorage(apnsCertPEM, apnsKeyPEM)
-					if err != nil {
-						initFatal(err, "initialize mdm apple MySQL storage")
-					}
-					depStorage, err = mds.NewMDMAppleDEPStorage([]byte(config.MDMApple.DEP.Token))
-					if err != nil {
-						initFatal(err, "initialize mdm apple dep storage")
-					}
-					nanoMDMLogger := NewNanoMDMLogger(kitlog.With(logger, "component", "apple-mdm-push"))
-					pushProviderFactory := buford.NewPushProviderFactory()
-					mdmPushService = nanomdm_pushsvc.New(mdmStorage, mdmStorage, pushProviderFactory, nanoMDMLogger)
+			// validate Apple BM config
+			if config.MDM.IsAppleBMSet() {
+				if !license.IsPremium() {
+					initFatal(errors.New("Apple Business Manager configuration is only available in Fleet Premium"), "validate Apple BM")
+				}
+
+				tok, err := config.MDM.AppleBM()
+				if err != nil {
+					initFatal(err, "validate Apple BM token, certificate and key")
+				}
+				appleBMToken = tok
+			}
+
+			if config.MDMApple.Enable {
+				if !config.MDM.IsAppleAPNsSet() || !config.MDM.IsAppleSCEPSet() {
+					initFatal(errors.New("Apple APNs and SCEP configuration must be provided to enable MDM"), "validate Apple MDM")
+				}
+
+				// TODO: for now (dogfood), Apple BM must be set when MDM is enabled,
+				// but when the MDM will be production-ready, Apple BM will be
+				// optional.
```

Comment:
just to make a personal note, are we tracking this somewhere?

Reply:
I created https://github.com/fleetdm/fleet/issues/8701 as a catch-all cleanup/tie loose ends ticket for when the MDM epic(s) would be done, I'll add that point there.